store/index.ts

```diff
@@ -10,14 +11,25 @@ import { HomeSectionState } from './homeSection';
 import { PlaybackManagerState } from './playbackManager';
 import { BackdropState } from './backdrop';
 import { DeviceState } from './deviceProfile';
-import { DisplayPreferencesApiState } from './displayPreferencesApi';
-import { CustomPreferences } from './settings';
+import { ClientSettingsState } from './clientSettings';
 import { websocketPlugin } from './plugins/websocket';
-import { preferencesSync } from './plugins/preferencesSync';
+import { preferencesSync } from './plugins/preferencesSyncPlugin';
+import { userPlugin } from './plugins/userPlugin';
 
-export const plugins = [websocketPlugin, preferencesSync];
+export const plugins = [websocketPlugin, preferencesSync, userPlugin];
+
+export interface AuthState {
+  busy: boolean;
+  loggedIn: boolean;
+  rememberMe: boolean;
+  strategy: string;
+  user: UserDto;
+}
 
 export interface RootState {
+  // A generic syncing indicator for settings or item syncing to and from the server
+  syncing: boolean;
```

Comment:
settingSyncing is more explicit imo. This could be anything.

Reply:
It is meant to be anything.

It's a generic sync indicator, to be used by clientSettings, playerSettings, fileSync or anything else that needs some amount of syncing.